renku/ui/service/controllers/api/mixins.py

```diff
@@ -169,8 +169,8 @@ def execute_op(self):
                     "git_url": self.request_data["git_url"],
                 }
 
-                if "ref" in self.request_data:
-                    clone_context["ref"] = self.request_data["ref"]
+                if "branch" in self.request_data:
+                    clone_context["ref"] = self.request_data["branch"]
```

Comment:
Shouldn't this be `clone_context["branch"]` (or include it too)? We don't use `"ref"` after this but we use `"branch"`.